File: kotlinlib/src/test/java/com/example/lib/DelegatedTest2_.java
package com.example.lib;

import org.jetbrains.annotations.NotNull;
import org.junit.jupiter.api.Test;

public class DelegatedTest2_ {
    @Test
    public final void main() {
        BaseImpl b = new BaseImpl(10);
        (new Derived(b)).printMessage();
    }

    public interface Base {
        void printMessage();

        void printMessageLine();
    }

    public static final class BaseImpl implements Base {
        private final int x;

        public BaseImpl(int x) {
            this.x = x;
        }

        @Override
        public void printMessage() {
            System.out.print(this.x);
        }

        @Override
        public void printMessageLine() {
            System.out.println(this.x);
        }

    }

    public static final class Derived implements Base {
        private final Base b;

        @Override
        public void printMessage() {
            String var1 = "abc";
            System.out.println(var1);
        }

        public Derived(@NotNull Base b) {
            this.b = b;
        }

        @Override
        public void printMessageLine() {
            this.b.printMessageLine();
        }
    }
}

File: kotlinlib/src/test/java/com/example/lib/DelegatedTest2.kt
package com.example.lib
import org.junit.jupiter.api.Test

class DelegatedTest2 {
    interface Base {
        fun printMessage()
        fun printMessageLine()
    }

    class BaseImpl(val x: Int) : Base {
        override fun printMessage() {
            print(x)
        }

        override fun printMessageLine() {
            println(x)
        }
    }

    class Derived(b: Base) : Base by b {
        override fun printMessage() {
            println("abc")
        }
    }

    @Test
    fun main() {
        val b = BaseImpl(10)
        Derived(b).printMessage()
    }

}